components/src/NavBar/DesktopMenu.js

```diff
@@ -6,29 +6,73 @@ import MenuTrigger from "./MenuTrigger";
 import MenuLink from "./MenuLink";
 import theme from "../theme";
 
-const isTrigger = menuItem => (menuItem.items);
+const ApplyMenuLinkStyles = styled.div({
+  "& *": {
+    display: "inline-flex",
+    color: theme.colors.white,
+    textDecoration: "none",
+    border: "none",
+    backgroundColor: "transparent",
+    justifyContent: "center",
+    alignItems: "center",
+    verticalAlign: "middle",
+    lineHeight: theme.lineHeights.base,
+    transition: ".2s",
+    fontSize: `${theme.fontSizes.medium}`,
+    padding: `${theme.space.x1} ${theme.space.x2}`,
+    borderRadius: theme.radii.medium,
+    "&:hover, &:focus": {
+      outline: "none",
+      color: theme.colors.lightBlue,
+      backgroundColor: theme.colors.black,
+      cursor: "pointer",
+    },
+    "&:disabled": {
+      opacity: ".5",
+    },
+  },
+});
+
+const renderMenuTrigger = menuItem => (
+  <div key={ menuItem.name }>
+    <MenuTrigger name={ menuItem.name } menuData={ menuItem.items } />
+  </div>
+);
+
+const renderMenuLink = menuItem => (
+  <div key={ menuItem.name }>
+    <MenuLink href={ menuItem.href }>
+      {menuItem.name}
+    </MenuLink>
+  </div>
+);
+
+const renderCustom = menuItem => (
+  <ApplyMenuLinkStyles key={ menuItem.name }>
+    {menuItem.render()}
+  </ApplyMenuLinkStyles>
+);
+
+const getRenderFunction = menuItem => {
+  if (menuItem.items) {
+    return renderMenuTrigger;
+  } else if (menuItem.href) {
+    return renderMenuLink;
+  } else if (menuItem.render) {
+    return renderCustom;
+  } else {
+    return (() => (null));
```

Comment:
I'm not sure if it's safe to include a `renderNothing` fallback here.
If there's nothing, it's an error, right?

I would consider raising an exception here:
`throw new Error("nothing to render")`?

I'm actually not sure if it's a good practice to throw errors inside render functions, so I would do a quick search for that. I did a 1 minute search, and didn't find much.

Reply:
Ideally this is handled with PropTypes. I've made a function to validate that menuData is correct in structure and content and throws errors if not.